src/QtUtils/MainThreadExecutorImplTest.cpp

```diff
@@ -154,4 +156,81 @@ TEST(MainThreadExecutorImpl, TryScheduleWithInvalidWeakPtr) {
   EXPECT_FALSE(result.has_value());
 }
 
-}  // namespace orbit_qt_utils
\ No newline at end of file
+TEST(MainThreadExecutorImpl, ScheduleAfterIfSuccessShortCircuitOnErrorVoid) {
+  auto executor = MainThreadExecutorImpl::Create();
+  bool called = false;
+  orbit_base::Promise<ErrorMessageOr<void>> promise{};
+  auto future = promise.GetFuture();
+  auto chained_future = executor->ScheduleAfterIfSuccess(future, [&called]() { called = true; });
+  EXPECT_FALSE(called);
+  EXPECT_FALSE(chained_future.IsFinished());
+
+  constexpr const char* const kErrorMessage{"Error"};
+  promise.SetResult(ErrorMessage{kErrorMessage});
+  QCoreApplication::processEvents();
+  EXPECT_FALSE(called);
+  EXPECT_TRUE(chained_future.IsFinished());
+  EXPECT_TRUE(chained_future.Get().has_error());
+  EXPECT_EQ(chained_future.Get().error().message(), kErrorMessage);
+}
```

Comment:
Add a test for a chain of 2 futures with ScheduleAfterIfSuccess?

Reply:
Good idea. Done.